include/common/comparator.h

```diff
@@ -0,0 +1,56 @@
+/* Copyright 2017 Pedro Cuadra <pjcuadra@gmail.com> & Meghadoot Gardi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. */
+
+#ifndef COMPARATOR_H_
+#define COMPARATOR_H_
+
+#include<systemc.h>
+
+/**
+ * Comparator Template
+ * @param width size of the output
+ */
+template<class T1, class T2>
+  SC_MODULE (comparator) {
```

Comment:
yes. we could extend it to have completely comparator. (But, we might not need so kept as "compare and return max")